packages/txservice/src/provider.ts

```diff
@@ -187,17 +251,17 @@ export class ChainRpcProvider {
    * @throws ChainError.reasons.ContractReadFailure in the event of a failure
    * to read from chain.
    */
-  public readTransaction(tx: ReadTransaction): ResultAsync<string, TransactionError> {
-    return this.resultWrapper<string>(true, async () => {
+  public async readTransaction(tx: ReadTransaction): Promise<string> {
+    return this.execute<string>(false, async (provider: SyncProvider) => {
       try {
-        return await this.provider.call(tx);
+        if (this.signer) {
+          this.signer.connect(provider);
+          return await this.signer.call(tx);
```

Comment:
should we just do `await this.signer.connect(provider).call(tx)` ?

Reply:
yes